packages/net/writebuffer.pony

```diff
@@ -0,0 +1,69 @@
+use "collections"
+
+class WriteBuffer
+  """
+  A simple constant sized byte buffer to push Numbers into.
+  The data container is iso in order to keep it sendable(?)
+  Potential upgrades:
+  1. Have a list of data buffers of preset size
+  2. Arguably, seperate functions for _be and _le
+  3. A better way to get data out of the buffer, without passing a new array?
+  """
+  var data: Array[U8] iso
+  var _offset: USize
+  var _little_endian: Bool
+
+  new create(size: USize, little_endian: Bool = false) =>
+    """
+    Allocate the buffer
+    """
+    data = recover Array[U8].undefined(size) end
+    _offset = 0
+    _little_endian = little_endian
+
+  fun ref get_data(x: Array[U8] iso): Array[U8] iso^ =>
+    """
+    Destructive read inorder to get an iso reference to data
+    """
+    var x': Array[U8] iso = data = consume x
+    consume x'
+
+  fun ref available(): USize =>
+    """
+    Number of bytes currently available
+    """
+    data.size() - _offset
+
+  fun ref _byte(value: U8) : WriteBuffer^ ? =>
+    """
+    Write a single byte into the buffer if space available else error
+    Update _offset
+    """
+    if _offset < data.size() then
+      data(_offset) = value
+      _offset = _offset + 1
+    else error end
+    this
+
+  fun ref write_number(value: Number): WriteBuffer? =>
```

Comment:
I don't necessarily like using a single method here for all number types.

I think a WriteBuffer is probably likely to be in a performance-sensitive part of an application, and if I understand the runtime correctly, this method will force the number values to be "boxed" so that their type can be detected at runtime, instead of being able to pass as raw numbers - meaning extra indirection and object allocations.

Also, with a method like this you can't pass number literals in directly, because the type is ambiguous - you'd have to do something like:
```pony
wb.write_number(U32(88))
```

at which point you might as well do this:

```pony
wb.write_u32(88)
```

you could even drop the `write_` prefix, since writing is what you do with a `WriteBuffer`. 

```pony
wb.u32(88)
``` 

And at this point, I'd say you might as well mirror how the `Buffer` methods look, with the `le`/`be` suffixes to make it explicit which byte order you're using.  I think the explicit byte order is important, and I think consistency with `Buffer` is important, given that these are in the same package and are intended to be conceptual mirror images of eachother.

```pony
wb.u32_le(88)
```